.github/workflows/compile-queries.yml

```
@@ -31,14 +31,17 @@ jobs:
           path: '*/ql/src/.cache'
           key: codeql-compile-pr-${{ github.sha }} # deliberately not using the `compile-compile-main` keys here.
           restore-keys: |
-            codeql-compile-main-${{ env.merge-base }}
-            codeql-compile-main-
+            codeql-compile-refs/heads/${{ github.base_ref }}-${{ env.merge-base }}
+            codeql-compile-refs/heads/${{ github.base_ref }}-
```

`github.base_ref` should already include the refs/heads prefix. Doesn't, it?

```suggestion
            codeql-compile-${{ github.base_ref }}-${{ env.merge-base }}
            codeql-compile-${{ github.base_ref }}-
```

---

No, it does not. I was also surprised by that.   

If you open the [latest compile-queries run](https://github.com/github/codeql/actions/runs/3427945326/jobs/5711566587) and expand `Read CodeQL query compilation - PR`, there you can see the cache keys that were computed.

---

Ah that is surprising, so `ref` is fully qualified but `base_ref` and `head_ref` are not.